libexec/basher-upgrade

```diff
@@ -14,6 +14,11 @@ if [ "$1" == "--complete" ]; then
   exec basher-list
 fi
 
+if [ "$1" == "--all" ]; then
+  basher-list | xargs -n1 basher upgrade
+  exit
```

Comment:
When we have tests we can check the behavior of xargs in practice, but I think in this case it will return 0 if at least one basher upgrade is successful, which means the whole script would return 0 even if some packages failed to upgrade.

Reply:
From the manpage:

```
xargs exits with the following status:

              0      if it succeeds

              123    if any invocation of the command exited with status 1-125

              124    if the command exited with status 255

              125    if the command is killed by a signal

              126    if the command cannot be run

              127    if the command is not found

              1      if some other error occurred.

       Exit codes greater than 128 are used by the shell to  indicate  that  a
       program died due to a fatal signal.
```

Reply:
this doesn't answer the question though. If 0 means every basher upgrade succeeded or at least one.

Reply:
```console
$ echo 'exit $1' > s
$ echo {0..10} | xargs -n1 bash s
$ echo $?
123
```